xdsl/ir.py

```diff
@@ -781,6 +761,18 @@ def verify(self, verify_nested_ops: bool = True) -> None:
             if isinstance(operand, ErasedSSAValue):
                 raise Exception("Erased SSA value is used by the operation")
 
+        if (parent_block := self.parent) and (parent_region := parent_block.parent):
+            # TODO single-block regions dealt when the NoTerminator trait is implemented
+            if len(parent_region.blocks) > 1:
+                if len(self.successors) > 0:
+                    if parent_block.last_op != self:
+                        raise Exception(
+                            "Operation with block successors must terminate its parent block"
+                        )
+
+                    if not self.has_trait(IsTerminator):
```

Comment:
This (`len(self.successors) > 0  -> has_trait(IsTerminator)`) should be checked even if the operation has no parents right?
Should we move this higher up?

Reply:
@math-fehr I ended up moving the condition for `has_trait(IsTerminator)` one indentation level since indeed there was an implicit assumption that a terminator must have successors, which is indeed not the case.

But my point regarding control flow (i.e., not belonging to a block/region) still stands; or am I missing something?

Reply:
Hmm that's a good question. It turns out that you cannot jump to the entry block, so technically it is not allowed.
We don't raise an error yet, the question is where should we raise an error.
I believe we should raise it here, as it reduce the amount of corner cases we have (otherwise, we rely on another verification pass to handle that case, and I feel this is very error-prone if that makes sense).
What do you think?